### PR TITLE
Some updates for the imaging experiments

### PR DESCRIPTION
**Pull request by MarcusZuber**
In the first commit I create a bit more code duplications, but this allows to do the actions in the _prepare_radios and _prepare_flats in a better order.

### CODE REVIEW

**Comment by MarcusZuber** [comment]
The state-checks we have to synchronize with https://github.com/ufo-kit/concert/pull/502, but the rest is open for discussion.
I did this mostly to make it easier to create sub-classes without rewriting the full `take_radios`.

**Comment by tfarago** [comment]
> The state-checks we have to synchronize with #502, but the rest is open for discussion. I did this mostly to make it easier to create sub-classes without rewriting the full `take_radios`.

Let's keep this about cleaning the actual DAQ things and #502 will take care of the state.

**Comment by MarcusZuber** [comment]
> > The state-checks we have to synchronize with #502, but the rest is open for discussion. I did this mostly to make it easier to create sub-classes without rewriting the full `take_radios`.
> 
> Let's keep this about cleaning the actual DAQ things and #502 will take care of the state.

The check fix is now only present for the imaging-experiments. For the base experiment and acquisition we use the commit from #502.

File: concert/experiments/imaging.py [hunk 1]
```diff
@@ -736,30 +750,8 @@ async def _finish_radios(self):
...
-            await self._prepare_radios()
-            await self._camera.set_trigger_source("SOFTWARE")
-            async with self._camera.recording():
-                for i in range(num_projections):
-                    await self._prepare_frame(i)
-                    await self._camera.trigger()
-                    yield await self._camera.grab()
-        finally:
-            await self._finish_radios()
-
```
Comment (tfarago) [inline concert/experiments/imaging.py:744]: both `_take_radios` slipped into this commit I guess.

**Review by tfarago — COMMENTED** [review]
A bad squash I guess. Otherwise this looks good.

File: concert/experiments/imaging.py (hunk 1 above)
Comment (MarcusZuber) [reply concert/experiments/imaging.py:744]: Do you like this commit msg more?

**Pull request merged**